tests/Foundation/FoundationTestResponseTest.php

```diff
@@ -232,6 +232,34 @@ public function testAssertJsonMissing()
         $response = TestResponse::fromBaseResponse(new Response(new JsonSerializableSingleResourceWithIntegersStub));
 
         $response->assertJsonMissing(['id' => 2]);
+
+        try {
+            $response->assertJsonMissing(['id' => 20]);
+            $this->fail('No exception was thrown');
```

Comment:
`$this->fail()` throws an exception, you should always put this outside of the try/catch

Reply:
These tests will pass even when `$response->assertJsonMissing(['id' => 20]);` doesn't throw an exception. The try catch will catch the exception `$this->fail()` throws

Reply:
The bulk of tests for this class are written like that.

Instead we should probably switch them to use:

```php
$this->expectException(\PHPUnit\Framework\AssertionFailedError::class);
```

See: https://phpunit.de/manual/6.5/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.exceptions

Reply:
The reason the other tests did it like that was because they had a seperate test to check if the test worked (which is very hacky): https://github.com/laravel/framework/blob/0ffdf3ed8d7e26a4ce3f30167e3840e3780a62d5/tests/Foundation/Testing/Constraints/SeeInOrderTest.php

I've submitted a PR that cleans up the `FoundationTestResponseTest` : https://github.com/laravel/framework/pull/25242